samza-core/src/main/java/org/apache/samza/container/grouper/task/GroupByContainerIds.java

```diff
@@ -89,7 +89,7 @@ public GroupByContainerIds(int count) {
     // Convert to a Set of ContainerModel
     Set<ContainerModel> containerModels = new HashSet<>();
     for (int i = 0; i < containerCount; i++) {
-      containerModels.add(new ContainerModel(containersIds.get(i), taskGroups[i]));
+      containerModels.add(new ContainerModel(containersIds.get(i), -1, taskGroups[i]));
```

Comment:
Not sure why here we are using -1 and in line 79 we are using 'i'? Better to be consistent. If there is a reason to use either '-1' or 'i', please add a comment to explain.